### PR TITLE
drivers/lwnl: add data pointer copy mode in lwnl_postmsg

### PR DESCRIPTION
**Pull request by an4967**
- add data pointer mode in lwnl_postmsg
(if data_len is negative, a queue copies pointer of data.)

### CODE REVIEW

File: os/net/blemgr/bledev_mgr_client.c [hunk 1]
```diff
@@ -66,7 +66,7 @@ static void bledrv_device_connected_cb(trble_device_connected *dev)
 
 static void bledrv_operation_notification_cb(trble_operation_handle *handle, trble_data *read_result)
 {
-	uint32_t size = sizeof(trble_conn_handle) + sizeof(trble_attr_handle) + sizeof(read_result->length) + read_result->length;
+	int32_t size = sizeof(trble_conn_handle) + sizeof(trble_attr_handle) + sizeof(read_result->length) + read_result->length;
```
Comment (jeongchanKim) [inline os/net/blemgr/bledev_mgr_client.c:69]: Is it okay not to check `read_result` is NULL or not?

File: os/drivers/lwnl/lwnl_evt_queue.c [hunk 2]
```diff
@@ -181,15 +181,20 @@ int lwnl_add_event(lwnl_cb_status type, void *buffer, uint32_t buf_len)
...
-			return -3;
+		if (buf_len < 0) {
+			evt->data.data = buffer;
+			evt->data.data_len = -(buf_len);
+		} else {
+			char *output = kmm_malloc(buf_len);
+			if (!output) {
+				LWNL_LOGE(TAG, "fail to alloc buffer");
+				kmm_free(evt);
+				return -3;
```
Comment (jeongchanKim) [inline os/drivers/lwnl/lwnl_evt_queue.c:192]: How about changing the return value between this line and line 206?

File: os/drivers/lwnl/lwnl_evt_queue.c [hunk 3]
```diff
@@ -181,15 +181,20 @@ int lwnl_add_event(lwnl_cb_status type, void *buffer, uint32_t buf_len)
 	evt->data.data = NULL;
 	evt->data.data_len = 0;
 	if (buffer) {
-		char *output = kmm_malloc(buf_len);
-		if (!output) {
-			LWNL_LOGE(TAG, "fail to alloc buffer");
-			kmm_free(evt);
-			return -3;
+		if (buf_len < 0) {
```
Comment (jeongchanKim) [inline os/drivers/lwnl/lwnl_evt_queue.c:184]: Could you leave comment about the case that `buf_len` is smaller than 0?

File: os/net/blemgr/bledev_mgr_client.c (hunk 1 above)
Comment (an4967) [reply os/net/blemgr/bledev_mgr_client.c:69]: 'read_result' is guranteed not to be NULL in BLE Stack layer.

File: os/drivers/lwnl/lwnl_evt_queue.c (hunk 3 above)
Comment (an4967) [reply os/drivers/lwnl/lwnl_evt_queue.c:184]: As I commented in this commit,  if data_len is negative, a queue copies pointer of data.

```suggestion
	trble_post_event(LWNL_EVT_BLE_CLIENT_NOTI, data, -(size));
```

**Review by pillip8282 — APPROVED** [review]
lgtm

**Pull request merged**